src/styles/main.scss

```diff
@@ -0,0 +1,65 @@
+$day-size: 100px;
+$day-gap: 1px;
+$calendar-padding: 10px;
+$calendar-column: 7;
+$calendar-width: $day-size * $calendar-column + $day-gap * ($calendar-column - 1) + 2 * $calendar-padding;
+$start-day: ('mon' 0, 'tue' 1, 'wed' 2, 'thu' 3, 'fri' 4, 'sat' 5, 'sun' 6);
+
+body {
+  margin: 0;
+  font-family: Arial, Helvetica, sans-serif;
+  display: flex;
+  justify-content: center;
+  align-items: center;
+  height: 100vh;
+}
+
+.calendar {
+  box-sizing: border-box;
+  display: flex;
+  padding: $calendar-padding;
+  width: $calendar-width;
+  gap: $day-gap;
+  flex-wrap: wrap;
+
+  &__day {
+    display: flex;
+    box-sizing: border-box;
+    width: $day-size;
+    height: $day-size;
+    background-color: #eee;
+    border: solid 1px #000;
+    position: relative;
+    transition: 0.5s;
+    
+    &:hover {
+      cursor: pointer;
+      transform: translateY(-20px);
+      background-color: #ffbfcb;
+    }
+  }
```

Comment:
The `&__day` element has a `transition` property (line 33) that applies to all changed properties. It is a good practice to specify which properties the transition is for, to avoid unintended effects and improve performance.